docs(rdbms-support-policy): update RDBMS version support policy with new database versions and deprecation rules (#7694)
camunda/camunda-docs: docs/self-managed/concepts/databases/relational-db/rdbms-support-policy.md

---
id: rdbms-support-policy
title: RDBMS version support policy
description: Defines Camunda’s official RDBMS support policy, including supported databases, LTS-based version rules, managed PostgreSQL guidance, JDBC driver expectations, and component compatibility.
---

Confirm which relational databases and versions Camunda supports for **Camunda 8 Self-Managed** and **Camunda 8 Run**, including lifecycle rules, managed PostgreSQL guidance, JDBC driver responsibilities, and component compatibility.

:::caution Work in progress
The RDBMS support policy is a work in progress. Content may change until finalized in a future alpha or minor release.
:::

## Scope and applicability

This policy applies to:

- **Camunda 8 Self-Managed** deployments.

> - **Camunda 8 Run**, where components support using an RDBMS as secondary storage.

Author: ```suggestion
```

description: Defines Camunda’s official RDBMS version support policy, including supported databases, version adoption rules, and JDBC driver guidance.
---

Camunda provides an official support policy for relational databases (RDBMS) used with **Camunda 8 Self-Managed**. This policy defines supported database vendors and versions, how new versions are adopted, how versions are deprecated and removed, how managed PostgreSQL services are treated, and how JDBC drivers are supported.

## Scope and applicability

This policy applies to **Camunda 8 Self-Managed** deployments.

It covers relational databases used for:

- [the secondary storage for the Orchestration Cluster](/self-managed/concepts/secondary-storage/index.md),
- [Web Modeler](/self-managed/components/modeler/web-modeler/configuration/database.md).

## Supported RDBMS

The following relational databases are officially supported:

| Database                 | Supported versions              |
| ------------------------ | ------------------------------- |
| PostgreSQL               | 14 (deprecated), 15, 16, 17, 18 |

> | Amazon Aurora PostgreSQL | 14 (deprecated), 15, 16, 17, 18 |

Reviewer: :question: Is Amazon Aurora PostgreSQL 18 already available for our runners and just not documented [here](https://confluence.camunda.com/spaces/HAN/pages/178590693/IAM+Roles+for+Service+Accounts+Testing+with+Aurora+and+OpenSearch+AWS+IRSA#IAMRolesforServiceAccountsTestingwithAuroraandOpenSearch(AWSIRSA)-AWSAuroraDB)? I am asking because we would need to add this version to our [cron-job](https://github.com/camunda/web-modeler/actions/runs/20705044335) which tests all supported database vendor versions.

Author: ✔️ @JacobiCamunda thanks for raising that! I just assumed there is version parity already but [it's not the case yet](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraPostgreSQLReleaseNotes/aurorapostgresql-release-calendar.html#aurorapostgresql.major.versions.supported). Removed ✔️

| MariaDB                  | 10.11, 11.4, 11.8               |
| MySQL                    | 8.4                             |
| Microsoft SQL Server     | 2019 (deprecated), 2022, 2025   |
| Oracle                   | 19c, 23ai                       |

> | H2                       | 2.3, 2.4                        |

Author: @christinaausley any objections to move forward with this direction?

> | H2                       | 2.3, 2.4                        |

Reviewer 2: I am good with this 👍

| H2                   | 2.x                     |

:::info
Changes to supported versions are announced in the [release notes](/reference/announcements-release-notes/890/890-release-notes.md).
:::

### Recommended database versions

For **new deployments**, standardize on **one of the latest two supported versions** of a given database whenever possible. Older supported versions remain valid for existing deployments but are not recommended for new deployments.

## Managed and compatible PostgreSQL services

Camunda supports **PostgreSQL as a database engine**, not individual managed service implementations.

PostgreSQL-compatible managed services (for example, AWS Aurora PostgreSQL or Azure Database for PostgreSQL) are expected to work if they are fully compatible with the supported PostgreSQL versions listed above.

This means:

- Camunda does not test or certify individual managed service implementations.
- Service-provider-specific operational behavior and service characteristics remain the responsibility of the service provider.
- If a managed service is fully compatible with a supported PostgreSQL version, it is considered supported from Camunda’s perspective.

:::note

> This support model is similar to Camunda’s Kubernetes support policy: Camunda supports the underlying technology and versions, while cloud providers are responsible for ensuring their managed offerings remain compatible.

Author: 🔧 Align with @theburi [comment](https://github.com/camunda/camunda-docs/pull/7694#discussion_r2675889915)

- **Availability window:** The version must be available at least three months before the next Camunda minor release.
- **Validation:** The version must be tested and validated across relevant Camunda components.

If the availability window is missed, support is deferred to a subsequent minor release.

:::note
Exceptions may be approved by engineering for strategic reasons or customer demand.
:::

## Version deprecation and removal

Versions are deprecated and removed using the following rules:

- **Vendor end of support (EoS):** Versions are deprecated when vendor support has ended or is scheduled to end within six months.
- **Removal:** Deprecated versions may be removed in a subsequent Camunda minor release once vendor support has ended.
- **Advance notice:** Deprecations and removals are communicated in advance via release notes.
- **Exceptions:** Support may be extended in exceptional cases, but any extension is explicitly documented and time-bound.

Camunda may remove support for a version if it contains known issues that prevent reliable operation.

## Database-specific support notes

### PostgreSQL

> Camunda supports multiple active PostgreSQL major versions concurrently based on the all LTS versions rule.

Author: PG don't have LTS versions - https://www.postgresql.org/developer/roadmap/

### MariaDB

Camunda supports **MariaDB LTS releases only**.

### MySQL

Camunda supports **MySQL LTS releases only**.

### Microsoft SQL Server

Camunda supports SQL Server versions that are in mainstream or extended vendor support.

### Oracle Database

Camunda supports **Oracle LTS releases**.

:::note
Oracle JDBC drivers are proprietary and must be provided by you at runtime.

> :::

Author: ```suggestion
```

Author: 🔧 I propose not to mix databases and drivers in here.

### H2

H2 is supported for development, testing, and evaluation only. Production use is not recommended.

## Supported JDBC driver versions

Camunda bundles JDBC drivers for databases where redistribution is permitted and expects you to provide drivers where licensing or distribution constraints apply (for example, Oracle).

Driver versions are not pinned as a formal support guarantee unless explicitly stated. Bundled and tested versions can change with dependency updates (for example, Spring Boot updates).

### Bundled drivers

The following JDBC drivers are included in the Camunda application images:

> | Database             | Driver artifact                        | Version       | Notes                                             |

Author: @stoerti do we also bundle the Aurora driver?

Reviewer 3: Yes, we bundle the `aws-advanced-jdbc-wrapper` in version 2.6.8.

```
      <dependency>
        <groupId>software.amazon.jdbc</groupId>
        <artifactId>aws-advanced-jdbc-wrapper</artifactId>
        <version>${version.aws-advanced-jdbc-wrapper}</version>
      </dependency>

```

Driver versions are not pinned. Use the latest driver version compatible with your database and environment.

For deployment instructions, see [loading JDBC drivers into pods](/self-managed/deployment/helm/configure/database/rdbms.md#loading-jdbc-drivers-into-pods).
:::

## JDBC driver policy

- Camunda supports the latest vendor-compatible JDBC driver for each supported RDBMS.
- You are responsible for providing JDBC drivers when required (for example, Oracle or MySQL).
- Driver versions are not pinned unless a specific version is required for compatibility or stability reasons.

## Component support matrix

This table shows RDBMS support status by component (including RDBMS as secondary storage where applicable):

| Component                         | Support status     | Notes                                                                                                             |
| :-------------------------------- | :----------------- | :---------------------------------------------------------------------------------------------------------------- |

> | **Orchestration Cluster (Zeebe)** | ✅ Fully supported | Supports RDBMS as secondary storage.                                                                              |

Author: ```suggestion
| **Orchestration Cluster** | ✅ Fully supported | Supports RDBMS as secondary storage.                                                                              |
```

| Tasklist                          | ✅ Fully supported | All functionality available.                                                                                      |
| Operate                           | ⚠️ Limited         | Under active development. Functionality is limited in **8.9 alpha**.                                              |

> | Optimize                          | ❌ Not supported   | Out of scope for **Camunda 8.9** RDBMS support.                                                                   |

Author: ```suggestion
| Optimize                          | ❌ Not supported   | Out of scope for RDBMS support.                                                                   |
```

| Web Modeler                       | ✅ Fully supported | See [Web Modeler database configuration](/self-managed/components/modeler/web-modeler/configuration/database.md). |
| Identity                          | ✅ Fully supported | All functionality available.                                                                                      |
| Management API (REST API)         | ✅ Fully supported | All functionality available.                                                                                      |

:::note

> "Orchestration Cluster (Zeebe)" refers to the orchestration runtime components. UI products are listed separately because their RDBMS support and maturity can differ by release.

Author: ```suggestion
"Orchestration Cluster" refers to the secondary storage of the Orchestration Cluster. UI products are listed separately because their RDBMS support and maturity can differ by the alpha release.
```
